src/plugins/core/sheet.ts

```diff
@@ -233,6 +235,36 @@ export class SheetPlugin extends CorePlugin<SheetState> implements SheetState {
     }
   }
 
+  /**
+   * Checks if all non-frozen header indices are present in the provided elements of selected rows/columns.
+   * This validation ensures that all rows or columns cannot be deleted when frozen panes exist.
+   */
+  private checkHeaderVisibility(
+    sheetId: UID,
+    dimension: Dimension,
+    elements: HeaderIndex[]
+  ): boolean {
+    const paneDivisions = this.getters.getPaneDivisions(sheetId);
+    const startIndex = dimension === "ROW" ? paneDivisions.ySplit : paneDivisions.xSplit;
+    const endIndex =
+      dimension === "ROW"
+        ? this.getters.getNumberRows(sheetId) - 1
+        : this.getters.getNumberCols(sheetId) - 1;
+
+    if (!startIndex) {
+      return false;
+    }
+
+    for (let i = startIndex; i <= endIndex; i++) {
+      if (!elements.includes(i)) {
+        // If the index is not present in the elements of selected rows/columns, then return false.
```

Comment:
This comment is kinda usless, you juste explain 2 easy to understand lines of code.